common/v2/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import { Address } from '@mycrypto/ui';
+import { truncate } from 'v2/utils';
+import { get3BoxProfile, ThreeBoxProfile } from 'v2/services';
+
+const AvatarImage = styled.img`
+  height: 100%;
+  width: 100%;
+  object-fit: cover;
+  border-radius: 50%;
+`;
+
+const Avatar = styled.div`
+  width: 60px;
+  height: 60px;
+`;
+
+interface Props {
+  address: string;
+  label?: string;
+}
+
+interface State {
+  profile: ThreeBoxProfile;
+}
+
+export default class Profile extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      profile: false
+    };
+  }
+
+  public componentDidMount() {
+    const { address } = this.props;
+    get3BoxProfile(address).then(result => {
+      this.setState({
+        profile: result
+      });
+    });
+  }
```

Comment:
When calling setState after an async call, make sure to cancel the request when the component un-mounts.
Otherwise we occasionally receive a cannot setState on unmounted Component error.
https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component
In the app we currently handle this case with common/v2/vendor/react-use/usePromise.ts, or any other https://github.com/streamich/react-use that is deemed useful.

See the use of `usePromise` in https://github.com/MyCryptoHQ/MyCrypto/pull/2947#discussion_r366251954